### PR TITLE
Editing2

### PR DESCRIPTION
**Pull request by Gandalfhfd**
I have parameterised all of the relevant tests.

### CODE REVIEW

File: e2e/editing.spec.ts [hunk 1]
```diff
@@ -98,146 +74,34 @@ test('Enable editing inputs', async ({ page }) => {
...
+const whitespaceType: string[] = ['		', '   ', ' 	 	'];
+let newTextWithWhitespace: string;
+const newTextWithoutWhitespace: string = 'Ipsum';
+const oldText: string = 'Lorem';
+
+for (const position of whitespacePosition) {
+    for (const i in descriptionOfWhitespaceType) {
+        test(`Remove ${position} ${descriptionOfWhitespaceType[i]} on save`, async ({ page }) => {
+
+            if(position === 'leading') {
```
Comment (KnightSky4) [inline e2e/editing.spec.ts:88]: Maybe switch...case instead?

File: e2e/editing.spec.ts [hunk 2]
```diff
@@ -98,146 +74,34 @@ test('Enable editing inputs', async ({ page }) => {
...
-
-    // Check that the leading whitespace has been removed when not in editing mode.
-    expect(await angularHomepage.checkTodoPresentByTextAndIsTrimmed(newTextWithWhitespace)).toBe(true);
-    // Check that the leading whitespace has been removed when in editing mode.
-    expect(await angularHomepage.checkTodoTrimmedInEditMode(newTextWithoutWhitespace)).toBe(true);
-});
+const whitespacePosition: string[] = ['leading', 'trailing', 'leading and trailing'];
+const descriptionOfWhitespaceType: string[] = ['tabs', 'spaces', 'tabs and spaces'];
+const whitespaceType: string[] = ['		', '   ', ' 	 	'];
+let newTextWithWhitespace: string;
```
Comment (KnightSky4) [inline e2e/editing.spec.ts:80]: Is this declaration needed outside the for loops?

File: e2e/editing.spec.ts [hunk 3]
```diff
@@ -98,146 +74,34 @@ test('Enable editing inputs', async ({ page }) => {
...
-
-    await angularHomepage.addNewTodo(oldText);
-    await angularHomepage.editTodo(oldText, newTextWithWhitespace, 'blur');
-
-    // Check that the leading whitespace has been removed when not in editing mode.
-    expect(await angularHomepage.checkTodoPresentByTextAndIsTrimmed(newTextWithWhitespace)).toBe(true);
-    // Check that the leading whitespace has been removed when in editing mode.
-    expect(await angularHomepage.checkTodoTrimmedInEditMode(newTextWithoutWhitespace)).toBe(true);
-});
+const whitespacePosition: string[] = ['leading', 'trailing', 'leading and trailing'];
```
Comment (KnightSky4) [inline e2e/editing.spec.ts:77]: Is it clearer to give these stronger types like on line 9?

Comment (Gandalfhfd) [reply e2e/editing.spec.ts:77]: I considered it, but don't see what it would add. I'm open to suggestions though

File: e2e/editing.spec.ts (hunk 2 above)
Comment (Gandalfhfd) [reply e2e/editing.spec.ts:80]: no, but the premature optimiser in me told me that it's faster this way. I'm happy to put it in the for loops

Comment (KnightSky4) [reply e2e/editing.spec.ts:80]: Should be fine as it is in that case

File: e2e/editing.spec.ts (hunk 3 above)
Comment (KnightSky4) [reply e2e/editing.spec.ts:77]: Only nicer tooltips on hover

Comment (Gandalfhfd) [reply e2e/editing.spec.ts:77]: Done!

**Review by KnightSky4 — DISMISSED** [review]

**Review by KnightSky4 — APPROVED** [review]

**Pull request merged**